### PR TITLE
added comments trap.[ch]

### PR DESCRIPTION
**Pull request by kim6515516**
Added comments as doxygen style into trcp.[ch]

### CODE REVIEW

**Comment by kesl** [comment]
Can one of the admins verify this patch?

**Comment by Wonseok** [comment]
s/added comments trap.[ch]/Added comments for trap.[ch]

File: hypervisor/hardware/arm32ve/libhw/trap.c [hunk 1]
```diff
@@ -7,21 +7,45 @@
 #include <vdev.h>
 #define DEBUG
 #include <log/print.h>
-
+/**\defgroup ARM terms.
```
Comment (Wonseok) [inline hypervisor/hardware/arm32ve/libhw/trap.c:10]: What is the ARM terms?

File: hypervisor/hardware/arm32ve/libhw/trap.c [hunk 2]
```diff
@@ -7,21 +7,45 @@
...
+ * <pre> ARM registers.
+ * ARM registers include 13 general purpose registers r0-r12, 1 Stack Pointer,
+ * 1 Link Register (LR), 1 Program Counter (PC).
+ * </pre>
+ */
+/**
+ * @brief Handles data abort exception taken from a mode other in Hyp mode.
+ * This fuction uses current ARM registers to dump and save as parameter.
+ * @param regs Current ARM registers used by current virtual guest.
```
Comment (Wonseok) [inline hypervisor/hardware/arm32ve/libhw/trap.c:19]: s/virtual guest/virtual machine

File: hypervisor/hardware/arm32ve/libhw/trap.c [hunk 3]
```diff
@@ -7,21 +7,45 @@
...
     guest_dump_regs(regs);
     hyp_abort_infinite();
     return HVMM_STATUS_UNKNOWN_ERROR;
 }
-
+/**
+ * @brief Handles IRQ exception when interrupt is occured by a device.
+ * This fucntion called gic interrupt and switched context.
+ * @param regs Current ARM registers used by current virtual guest.
+ * For more information, check line 10 in this file(defgroup ARM terms).
```
Comment (Wonseok) [inline hypervisor/hardware/arm32ve/libhw/trap.c:33]: See the line 10.

**Pull request closed without merging**